src/main/kotlin/org/rust/cargo/project/model/CargoProjectService.kt

```diff
@@ -28,11 +28,97 @@ import java.util.concurrent.CompletableFuture
 import java.util.concurrent.TimeUnit
 
 /**
- * [CargoProjectsService] stores a list of `Cargo.toml` file,
- * registered with the current IDE project. Each `Cargo.toml`
- * is represented by a [CargoProject], whose main attribute is
- * `workspace`: a description of a Cargo project acquired from
- * Cargo itself via `cargo metadata` command.
+ * Stores a list of [CargoProject]s associated with the current IntelliJ [Project].
+ * Use [Project.cargoProjects] to get an instance of the service.
+ *
+ * # Project model
+ *
+ * ```
+ *                   [CargoProject]
+ *                         |
+ *                 [CargoWorkspace]
+ *                   /          \
+ *              [Package]   [Package]
+ *                /   \         |
+ *         [Target] [Target] [Target]
+ *        (main.rs) (lib.rs) (lib.rs)
+ * ```
+ *
+ * ## CargoProject
+ *
+ * [CargoProject] is basically a `Cargo.toml` file inside a project source folder
+ * that is linked to the current IDE project via [attachCargoProject] (usually
+ * there is a single project). Each valid [CargoProject] should contain
+ * exactly one [CargoWorkspace] (see [CargoProject.workspace]). A workspace
+ * can be null if project is not valid (project is in updating state, Cargo is
+ * not installed, broken `Cargo.toml`, etc).
+ *
+ * ## CargoWorkspace
+ *
+ * [CargoWorkspace] is attached to each valid [CargoProject] and stores info
+ * about its packages (see [CargoWorkspace.packages]). A workspace is acquired
+ * from Cargo itself via `cargo metadata` command.
+ *
+ * ## Package
+ *
+ * [CargoWorkspace.Package] is a thing that can be considered as a dependency (has
+ * a name and version) and can have dependencies. A package can consist of multiple
+ * binary (executable) targets and/or one library target. Package is described by
+ * `[package]` section of `Cargo.toml`
+ *
+ * ## Target
+ *
+ * [CargoWorkspace.Target] is a thing that can be compiled to some binary artifact,
+ * e.g. executable binary, library, example library/executable, test executable.
+ * Each target has crate root (see [CargoWorkspace.Target.crateRoot]) which is
+ * usually `main.rs` or `lib.rs`.
```

Comment:
Let's replace "usually `main.rs` or `lib.rs`" to "for example, `main.rs` or `lib.rs`" because you may have one `main.rs`, one `lib.rs` and a lot of crate roots for test targets